### PR TITLE
Use Assets or Resources based on platform.

### PR DESCRIPTION
**Pull request by yschimke**
Assets uncompressed on Android. Assuming that the AAR file will compress.

Use Resources, compressed on JVM.

Includes ~an Androidx Startup Initializer~ a ContentProvider to grab the app context.

### CODE REVIEW

**Comment by yschimke** [comment]
Androidx Startup approach fails with

https://github.com/square/okhttp/actions/runs/12520208963/job/34925534823?pr=8636

```
Execution failed for task ':okhttp:animalsnifferMain'.
> Could not resolve all files for configuration ':okhttp:compileClasspath'.
   > Could not resolve androidx.startup:startup-runtime:1.2.0.
     Required by:
         project :okhttp
      > No matching variant of androidx.startup:startup-runtime:1.2.0 was found. The consumer was configured to find a library for use during compile-time, compatible with Java 8, preferably in the form of class files, preferably optimized for standard JVMs, and its dependencies declared externally but:
```

File: okhttp/build.gradle.kts [hunk 1]
```diff
@@ -3,10 +3,9 @@
 import aQute.bnd.gradle.BundleTaskExtension
 import com.vanniktech.maven.publish.JavadocJar
 import com.vanniktech.maven.publish.KotlinMultiplatform
-import org.gradle.kotlin.dsl.compileOnly
+import org.gradle.internal.fingerprint.classpath.impl.ClasspathFingerprintingStrategy.compileClasspath
```
Comment (yschimke) [inline okhttp/build.gradle.kts:6]: oops

**Comment by JakeWharton** [comment]
androidx.startup is the right choice, though. The cost of each content provider is non-zero. Animal sniffer needs changed to run on the extracted jars from the aars, or just dropped from use on the Android target.

**Comment by yschimke** [comment]
@JakeWharton trying https://github.com/gmazzo/gradle-aar2jar-plugin

**Pull request merged**